File: src/net/eduard/api/command/map/MapLoadCommand.java

package net.eduard.api.command.map;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import net.eduard.api.lib.Mine;
import net.eduard.api.lib.manager.CommandManager;

public class MapLoadCommand extends CommandManager {

	public MapLoadCommand() {
		super("load", "carregar");
		setDescription("Carrega um mundo descarregado");
	}

	
	@Override
	public boolean onCommand(CommandSender sender, Command command,
			String label, String[] args) {
		if (args.length == 1) {
			sender.sendMessage("§c/map load <name>");
		} else {
			if (Mine.onlyPlayer(sender)) {
				Player p = (Player) sender;
				if (Mine.MAPS.containsKey(args[1].toLowerCase())) {
					Mine.MAPS_CACHE.put(p, Mine.MAPS.get(args[1].toLowerCase()));
					p.sendMessage("§bEduardAPI §aMapa carregado com sucesso!");
				} else {
					p.sendMessage("§bEduardAPI §aMapa invalido: §2" + args[1]);
				}

			}
		}

		return true;
	}

}

File: src/main/java/net/eduard/api/command/map/MapLoadCommand.kt
package net.eduard.api.command.map

import net.eduard.api.lib.manager.CommandManager
import net.eduard.api.lib.modules.Mine
import net.eduard.api.server.minigame.MinigameSchematic
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class MapLoadCommand : CommandManager("load", "carregar") {
    override fun command(sender: CommandSender, args: Array<String>) {
        if (args.isEmpty()) {
            sendUsage(sender)
            return
        }
        if (Mine.onlyPlayer(sender)) {
            val player = sender as Player
            val name = args[0].toLowerCase()
            if (MinigameSchematic.exists(name)) {
                MinigameSchematic.loadToCache(player, name)
                player.sendMessage("§bEduardAPI §aMapa carregado com sucesso!")
            } else {
                player.sendMessage("§bEduardAPI §aMapa invalido: §2" + args[1])
            }
        }

    }

    init {
        usage = "/map load <name>"
        description = "Carrega um mundo descarregado"
    }
}